File: 3151-special-array-i/3151-special-array-i.java
class Solution {
    public boolean isArraySpecial(int[] nums) {
        if(nums.length<=1){
            return true;
        }
        boolean flag = false;
        for(int i=0; i<nums.length-1; i++){
            if((nums[i]%2==0 && nums[i+1]%2!=0)
            ||
            (nums[i]%2!=0 && nums[i+1]%2==0)){
                flag = true;
            }else{
                return false;
            }
        }
        return flag;
    }
}

File: 3151-special-array-i/3151-special-array-i.cs
public class Solution {
    public bool IsArraySpecial(int[] nums) {
        for (int i = 0; i < nums.Length - 1; i++)
            if (nums[i] % 2 + nums[i + 1] % 2 != 1)
                return false;
        return true;
    }
}